karabo_data/geometry2.py

```diff
@@ -335,6 +335,68 @@ def to_distortion_array(self):
 
         return distortion
 
+    def _tile_slice(self, tileno):
+        """Implement in subclass: which part of module array each tile is.
+        """
+        raise NotImplementedError
+
+    def get_pixel_positions(self, centre=True):
+        """Get the coordinates of each pixel in the detector
+
+        The output is an array with shape like the data, with an extra dimension
+        of length 3 to hold (x, y, z) coordinates. Coordinates are in pixel
+        units from the detector centre.
+
+        If centre=True, the coordinates are calculated for the centre of each
+        pixel. If not, the coordinates are for the first corner of the pixel
+        (the one nearest the [0, 0] corner of the tile in data space.
+        """
+        out = np.zeros(self.expected_data_shape + (3,), dtype=np.float32)
```

Comment:
Wouldn't it be better to have a 2D array as input, e.g an array of definitions of multiple ROI's, and map that 2D array onto the 3D array for (nmodules x slow_scan x fast_scan). I know this is quite different to the approach below but maybe worthwhile considering?

Reply:
I understand the desire to 'disassemble' an image, and I can implement that if required. But I think this approach is actually quite powerful - you can use the full precision and 3D information of the available geometry information to define a mask.

This kind of question is why I'm waiting for Henry's input before adding tests and docs for what I've implemented.

Reply:
Ok good, we'll keep that in mind at leave it as it is for now